### PR TITLE
add wine, rcedit, update to focal

### PR DESCRIPTION
**Pull request by monban**
This PR does the following:

- Updates Ubuntu to focal, by using the latest version of git we can use the [Checkout action v2](https://github.com/marketplace/actions/checkout)
- Installs wine, this is required to use rcedit
- Downloads rcedit to /usr/loca/bin, required to change / correct the icon on build Windows executables
- Makes downloads quiet, less spam in the build log

_Contributor Edit:_
Fixes https://github.com/aBARICHELLO/godot-ci/issues/21

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
+    wine-stable \
     && rm -rf /var/lib/apt/lists/*
 
+RUN wget --quiet https://github.com/electron/rcedit/releases/download/v1.1.1/rcedit-x64.exe -O /usr/local/bin/rcedit.exe
+
 ENV GODOT_VERSION "3.2.3"
 
-RUN wget https://downloads.tuxfamily.org/godotengine/${GODOT_VERSION}/Godot_v${GODOT_VERSION}-stable_linux_headless.64.zip \
-    && wget https://downloads.tuxfamily.org/godotengine/${GODOT_VERSION}/Godot_v${GODOT_VERSION}-stable_export_templates.tpz \
+RUN wget --quiet https://downloads.tuxfamily.org/godotengine/${GODOT_VERSION}/Godot_v${GODOT_VERSION}-stable_linux_headless.64.zip \
```
Comment (asheraryam) [inline Dockerfile:25]: I like this "spam in the build log", it helps make sure the download path didn't break upstream. Does it still show errors with the quiet flag? If so then it's fine.

**Review by asheraryam — APPROVED** [review]

File: Dockerfile (hunk 1 above)
Comment (Calinou) [reply Dockerfile:25]: `curl -fsSL` is typically what I use with curl (quiet normally, but displays errors and follows redirects).

Comment (monban) [reply Dockerfile]: After checking an observing that `--quiet` does indeed suppress, for example, 404s, I've changed it to `--no-verbose`. I agree this is much better functionality.

**Comment by monban** [comment]
As per the conversation above, I've changed `--quiet` to `--no-verbose`. Also I'm wondering if it would be a good idea to document where as to how to use rcedit? Here is roughly what I'm doing on github actions:
```
      - name: Setup
        run: |
          chown -R root:root $HOME
          mkdir -v -p ~/.local/share/godot/templates
          mv /root/.local/share/godot/templates/${GODOT_VERSION}.stable ~/.local/share/godot/templates/${GODOT_VERSION}.stable
      - name: Windows Desktop Build
        run: |
          mkdir -v -p $GITHUB_WORKSPACE/build/windows-desktop
          cd $GITHUB_WORKSPACE/src
          godot -v --export "Windows Desktop" $GITHUB_WORKSPACE/build/windows-desktop/$EXPORT_NAME.exe
      - name: Convert png to ico
        run: convert $GITHUB_WORKSPACE/src/assets/common/textures/icons/icon.png -define icon:auto-resize="256,128,96,64,48,32,16" ~/icon.ico
      - name: Change exe icon
        run: cd $GITHUB_WORKSPACE; wine $GITHUB_WORKSPACE/rcedit.exe $GITHUB_WORKSPACE/build/windows-desktop/$EXPORT_NAME.exe --set-icon ~/icon.ico
```
In any event, please re-review and let me know if I can help by touching up the documentation anywhere.

**Comment by asheraryam** [comment]
@monban If you'd like you can add a note at the bottom of README and/or a comment to the gitlab and github config scripts.

**Comment by aBARICHELLO** [comment]
[Pipelines are working](https://gitlab.com/BARICHELLO/godot-ci/-/pipelines/207459332)

**Review by aBARICHELLO — CHANGES_REQUESTED** [review]
I agree with @asheraryam, a short 'how to use rcedit' section in the readme would be helpful

**Comment by aBARICHELLO** [comment]
Using wine for the rcedit with GitHub actions is failing with this error message:
`wine: '/__w/godot-ci/godot-ci' is not owned by you, refusing to create a configuration directory there`
Action log: https://github.com/aBARICHELLO/godot-ci/runs/1371907817?check_suite_focus=true

Also [act](https://github.com/nektos/act) is giving me different results locally, testing this is a real pain.

**Comment by monban** [comment]
Yeah, unfortunately you need to chown the home directory when running from GithubActions, [here is an example of usage](https://github.com/opensuspect/opensuspect/blob/main/.github/workflows/main.yml), I've been trying to figure out if there is a better way to accomplish this, hence why I haven't created the above referenced documentation yet...

**Comment by monban** [comment]
(note the chown on [line 30](https://github.com/opensuspect/opensuspect/blob/4854e107bba7119b6d49c00f849002ec4c15301c/.github/workflows/main.yml#L30))

**Comment by 2shady4u** [comment]
@monban do you have any updates on this?

**Comment by monban** [comment]
You know what, I completely forgot about this, I'm sorry. I'll get on this after work today, expect an update in the next couple of hours.

**Comment by Deozaan** [comment]
This looks really useful. Is there any chance of this getting updated and merged? @monban

**Comment by realkotob** [comment]
@Deozaan If you are interesting in picking this up feel free to open a new PR with the changes.

**Comment by Deozaan** [comment]
> @Deozaan If you are interesting in picking this up feel free to open a new PR with the changes.

I've looked into it briefly and I think it's beyond my skill level. 😳😬

**Comment by monban** [comment]
At this point I no longer recommend people to use rcedit, but rather to use [Godot Icon](https://github.com/pkowal1982/godoticon) instead.

**Comment by Calinou** [comment]
Superseded by https://github.com/abarichello/godot-ci/pull/146 .

**Pull request closed without merging**